internal/controllers/clusterextension_controller.go

```diff
@@ -206,6 +207,11 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1.Cl
 	installedBundle, err := r.InstalledBundleGetter.GetInstalledBundle(ctx, ext)
 	if err != nil {
 		setInstallStatus(ext, nil)
+		var saerr *tokengetter.ServiceAccountNotFoundError
+		if errors.As(err, &saerr) {
+			l.Info("is a SAError")
+			err = saerr
+		}
```

Comment:
Why do we need to log something here?
Also, we are checking the error and replacing it 
Why do we need it?

Reply:
we are checking the type of the error and then returning our error message